### ISSUE TITLE
How to load Vulkan on macOS?

### ISSUE THREAD

**Issue by ghostman2013** [KyleMayes/vulkanalia#255]
Hello!

[This tutorial](https://github.com/KyleMayes/vulkanalia/blob/master/tutorial/src/01_instance_creation.rs) says to load Vulkan library as below:  
```
let loader = LibloadingLoader::new(LIBRARY)?;
let entry = Entry::new(loader).map_err(|b| anyhow!("{}", b))?;
```
Unfortunately, this code doesn't work on **macOS 14 Sonoma** and **Vulkan SDK v1.3.275.0**, there are just no `libvulkan.dylib` in paths where the **libloading** is searching. However, it's not a problem, I just copied libraries to a project folder.

The problem is that Vulkan library requires Molten VK on macOS. It can be specified using [icd.json](https://vulkan.lunarg.com/doc/sdk/latest/mac/getting_started.html) but I can't get how to provide `icd.json` and `libmoltenvk.dylib` for **vulkanalia** library.

Is there any sample how to do that?

**Comment by KyleMayes** [comment]
## tl;dr

Run `source ~/VulkanSDK/1.3.275.0/setup-env.sh` (or update to match the Vulkan SDK version you are using) before attempting to run a Vulkan app to set up the necessary paths. I added a similar statement to my `~/.zshrc`.

## Findings

I tested `cargo run --bin triangle` on my M2 MacBook air running macOS 14.2.1.

I had an older Vulkan SDK installed (1.3.243.0) which worked fine.

I uninstalled that and installed the latest Vulkan SDK (1.3.280.1).
After that, `cargo run --bin triangle` was no longer working since it couldn't find `libvulkan.dylib`.

Then I remembered I had added this to my `~/.zshrc`:

```
source ~/VulkanSDK/1.3.243.0/setup-env.sh
```

I updated that to point to the new Vulkan SDK I had just installed, ran `source ~./zshrc`, and now `cargo run --bin triangle` was running properly again.

So I think macOS users will do something similar (run `setup-env.sh` from the Vulkan SDK).

I don't remember when I did this originally, but I should probably add information about this to the tutorial chapter on installing the SDK, as well as maybe the FAQ chapter and the `README.md`.

**Comment by KyleMayes** [comment]
Added usage of `setup-env.sh` script for macOS to Vulkan SDK setup instructions in tutorial and also added a reference to that new section in the FAQ.

**Comment by ghostman2013** [comment]
> Added usage of `setup-env.sh` script for macOS to Vulkan SDK setup instructions in tutorial and also added a reference to that new section in the FAQ.

Thank you for the answer! 

Sorry, it's a little off-topic but if I need to use a local libraries (copied to the project folder **libvulkan.dylib** and **libMoltenVK.dylib**)... Is it possible to link them correctly to **vulkanalia**? I couldn't find where to put the **icd.json** in **/target** folder to a local Vulkan library could find a local MoltenVK lib.
